indico/modules/events/models/events.py

```diff
@@ -521,6 +521,11 @@ def move_start_dt(self, start_dt):
             entry.move(new_dt)
         self.start_dt = start_dt
 
+    def move(self, category):
+        if self.category != category and not category.children:
```

Comment:
This check doesn't really belong in here IMO. I'd unconditionally assign the new category here (and add a TODO comment after the flush to trigger the moved signal)